File: exerc-3/sort.cpp
#include <iostream>
#include <vector>
#include <string>
#include <bits/stdc++.h>

class Item{
    public:

    std::string Name;
    int Key;

    //construção
    Item(std::string name, int key){
        Name = name;
        Key = key;
    }
};

//vetor com itens:
std::vector<Item> array ={
    {"bill", 3},
    {"neil", 4},
    {"john", 2},
    {"rick", 5},
    {"alex", 1},
};

//representação dos itens da array
void repr(const std::vector<Item>array){
    for (auto &e: array){
        std:: cout << "Sort Item -> Nome: " << e.Name << ", key:" << e.Key << std::endl;
        //std:: cout << "\n";
    }
};

//ordenação dos itens
bool ordem(Item& a, Item& b){
    return (a.Key < b.Key);
}

int main(void){

    std::sort(array.begin(), array.end(), ordem);
    repr(array);

return 0;
}

File: exerc-3/sort.py
class Item:
    def __init__(self, nome:str, key: int):
        self.nome = nome
        self.key = key

    def __repr__(self):
        return f'Item -> Nome: {self.nome}, key: {self.key} \n'

def ordem(k):
    return k.key

Itens = [
    Item("bill", 3),
    Item("neil", 4),
    Item("john", 2),
    Item("rick", 5),
    Item("alex", 1)
]

Itens.sort(key = ordem)
print(Itens)
